### PR TITLE
Add for each node to basic flink nodes components

### PR DESCRIPTION
**Pull request by kdutkowski**

### CODE REVIEW

File: engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/RequestResponseInterpreterSpec.scala [hunk 1]
```diff
@@ -105,7 +105,7 @@ class RequestResponseInterpreterSpec extends FunSuite with Matchers with Patient
     val process = EspProcessBuilder
       .id("proc1")
       .source("start", "request1-post-source")
-      .customNode("for-each", "outPart", "for-each", "elements" -> "#input.toList()")
+      .customNode("for-each", "outPart", "for-each", "Elements" -> "#input.toList()")
```
Comment (dswiecki) [inline engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/RequestResponseInterpreterSpec.scala:108]: can we do it without migration ?

Comment (mproch) [reply engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/RequestResponseInterpreterSpec.scala:108]: we already changed component name... it's not (widely) used in production...

File: engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    extractResultValues(results) shouldBe List("one_1", "other_1")
+  }
+
+  test("should not produce any results when elements list is empty") {
+    val model = modelData(List(TestRecord()))
+
+    val testProcess = aProcessWithForEachNode(elements = "{}")
+
+    val results = collectTestResults[String](model, testProcess)
+    results.nodeResults.asJava shouldNot contain key sinkId
```
[the remark this answers is not in the document]
Comment (kdutkowski) [reply engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala:46]: good point, removing it

File: engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  def modelData(list: List[TestRecord] = List()): LocalModelData = LocalModelData(ConfigFactory
+    .empty().withValue("useTypingResultTypeInformation", fromAnyRef(true)), new Creator(list))
+
+  private def aProcessWithForEachNode(elements: String, resultExpression: String = s"#$forEachOutputVariableName") =
+    EspProcessBuilder
+      .id("forEachProcess")
+      .parallelism(1)
+      .stateOnDisk(true)
+      .source("start", "start")
+      .buildSimpleVariable("id", "id", "#input.id")
```
[the remark this answers is not in the document]
Comment (kdutkowski) [reply engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala:58]: no, we don't

File: engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/RequestResponseInterpreterSpec.scala (hunk 1 above)
Comment (kdutkowski) [reply engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/RequestResponseInterpreterSpec.scala:108]: that's what I hoped for, the component is only available in lite engine, that has not been released yet.

File: engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  private def extractResultValues(results: TestProcess.TestResults[Any]): List[String] = results.nodeResults(sinkId)
+    .map(_.variableTyped[String](resultVariableName).get)
+
+  private def runProcess(model: LocalModelData, testProcess: EspProcess, collectingListener: ResultsCollectingListener): Unit = {
+    val stoppableEnv = flinkMiniCluster.createExecutionEnvironment()
+    val registrar = FlinkProcessRegistrar(new FlinkProcessCompiler(model) {
+      override protected def listeners(processObjectDependencies: ProcessObjectDependencies): Seq[ProcessListener] =
+        List(collectingListener) ++ super.listeners(processObjectDependencies)
```
[the remark this answers is not in the document]
Comment (kdutkowski) [reply engine/flink/components/base/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/ForEachTransformerSpec.scala]: Turns out we can :) Take a look now, I changed it so it's initialised at the beginning of each test.

**Review by mproch — APPROVED** [review]

**Pull request merged**